iocontrol/iocontrol.go

```diff
@@ -1,13 +1,190 @@
 package iocontrol
 
-import "go.uber.org/zap"
+import (
+	"go.uber.org/zap"
 
+	"github.com/macformula/hil/iocontrol/raspi"
+	"github.com/macformula/hil/iocontrol/speedgoat"
+)
+
+const (
+	_loggerName = "iocontrol"
+)
+
+// IOControlOption is a type for functions operating on IOControl
+type IOControlOption func(*IOControl)
+
+// IOControl contains I/O controllers
 type IOControl struct {
+	sg *speedgoat.Controller
+	rp *raspi.Controller
+
 	l *zap.Logger
 }
 
-func NewIOControl(l *zap.Logger) *IOControl {
-	return &IOControl{
-		l: l,
+// NewIOControl returns a new IOControl
+func NewIOControl(
+	l *zap.Logger,
+	opts ...IOControlOption) *IOControl {
+	io := &IOControl{
+		sg: nil,
+		rp: nil,
+
+		l: l.Named(_loggerName),
+	}
+
+	for _, o := range opts {
+		o(io)
+	}
+
+	return io
+}
+
+// WithSpeedgoat initializes the iocontroller with a speedgoat device
+func WithSpeedgoat(sg *speedgoat.Controller) IOControlOption {
+	return func(i *IOControl) {
+		i.sg = sg
+	}
+}
+
+// WithRaspi WithRapsi initializes the iocontroller with a raspi device
+func WithRaspi(rp *raspi.Controller) IOControlOption {
+	return func(i *IOControl) {
+		i.rp = rp
+	}
+}
+
+// SetDigital sets an output digital pin for a specified pin
+func (I *IOControl) SetDigital(output DigitalPin, b bool) error {
```

Comment:
Change receiver to be lower case. I think it would be acceptable to name the receiver `io` in this case.
```suggestion
func (io *IOControl) SetDigital(output DigitalPin, b bool) error {
```